tests/table.spec.ts

```diff
@@ -140,6 +140,179 @@ test.describe('tests complex table operations', () => {
+
+    // Wait until it's done loading
+    await expect(page.locator('.iris-grid-panel .loading-spinner')).toHaveCount(
+      0
+    );
+
+    // Model is loaded, need to make sure table data is also loaded
+    await expect(
+      page.locator('.iris-grid .iris-grid-loading-status')
+    ).toHaveCount(0);
+
+    const tableOperationsMenu = page.locator(
+      'data-testid=btn-iris-grid-settings-button-table'
+    );
+    await tableOperationsMenu.click();
+
+    await expect(page.locator('.table-sidebar')).toHaveCount(1);
+
+    await page.locator('data-testid=menu-item-Organize Columns').click();
+
+    await test.step('Search', async () => {
+      await page.getByPlaceholder('Search').click();
+      await page.keyboard.type('d');
+
+      await expect(
+        page.locator('.visibility-ordering-builder')
+      ).toHaveScreenshot();
+    });
+
+    await test.step('Hide Selected', async () => {
+      await page.getByRole('button', { name: 'Hide Selected' }).click();
+
+      await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+    });
+
+    await test.step('Move Selection Down', async () => {
+      await page.getByRole('button', { name: 'Move selection down' }).click();
+
+      await expect(
+        page.locator('.visibility-ordering-builder')
+      ).toHaveScreenshot();
+    });
+
+    await test.step('Move Selection Up', async () => {
+      await page.getByRole('button', { name: 'Move selection up' }).click();
+
+      await expect(
+        page.locator('.visibility-ordering-builder')
+      ).toHaveScreenshot();
+    });
+
+    await test.step('Move Selection to Bottom', async () => {
+      await page
+        .getByRole('button', { name: 'Move selection to bottom' })
+        .click();
+
+      await expect(
+        page.locator('.visibility-ordering-builder')
+      ).toHaveScreenshot();
+    });
+
+    await test.step('Move Selection to Top', async () => {
+      await page.getByRole('button', { name: 'Move selection to top' }).click();
+
+      await expect(
+        page.locator('.visibility-ordering-builder')
+      ).toHaveScreenshot();
+    });
+
+    await test.step('Sort Descending', async () => {
+      await page.getByRole('button', { name: 'Sort descending' }).click();
+
+      await expect(
+        page.locator('.visibility-ordering-builder')
+      ).toHaveScreenshot();
+    });
+
+    await test.step('Sort Ascending', async () => {
+      await page.getByRole('button', { name: 'Sort ascending' }).click();
+
+      await expect(
+        page.locator('.visibility-ordering-builder')
+      ).toHaveScreenshot();
+    });
+
+    await test.step('Reset', async () => {
+      await page.getByRole('button', { name: 'Reset' }).click();
+
+      await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+      await expect(
+        page.locator('.visibility-ordering-builder')
+      ).toHaveScreenshot();
+    });
+
+    // TO DO: Mimic drag function
```

Comment:
For TODOs, you should just do one word `TODO`, and a ticket number is required if you're going to merge in the TODO, or you can resolve it before merging. e.g. :
```
 // TODO: #123 Need to follow up on this
```